tools/ruff_baselined.py

```
@@ -0,0 +1,350 @@
+def run_ruff(paths: list[str]) -> list[Violation]:
+    cmd = [
+        ".venv/bin/ruff",
+        "check",
+        "--select",
+        ",".join(RULES),
+        "--output-format",
+        "concise",
+        "--no-cache",
+        *paths,
+    ]
+    result = subprocess.run(cmd, cwd=REPO_ROOT, capture_output=True, text=True)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)